src/store/store.ts

```diff
@@ -1433,8 +1433,13 @@ export const useStore = defineStore('store', {
     setWindowWidth (width: number) {
       this.stateModel.windowWidth = width
     },
+
+    //
+    // alamgamation-specific setters
+    //
     setAmalgamatingBusinesses (amalgamatingBusinesses: Array<AmalgamatingBusinessIF>) {
       this.stateModel.amalgamation.amalgamatingBusinesses = amalgamatingBusinesses
+      if (!this.stateModel.ignoreChanges) this.stateModel.haveChanges = true
```

Comment:
"If the user has changed something then set a flag so we can try to stop users from exiting without saving their changes."

There were quite a few of these missing. I only updated some amalgamation and restoration setters.